wgengine/magicsock/endpoint.go

```diff
@@ -415,22 +416,28 @@ func (de *endpoint) cliPing(res *ipnstate.PingResult, size int, cb func(*ipnstat
 		return
 	}
 
-	de.pendingCLIPings = append(de.pendingCLIPings, pendingCLIPing{res, cb})
```

Comment:
Instead of moving this to startDiscoPingLocked, perhaps startDiscoPingLocked should just return txid that this function will put into pendingCLIPing?